presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowErrorCode.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.plugin.arrow;
+
+import com.facebook.presto.common.ErrorCode;
+import com.facebook.presto.common.ErrorType;
+import com.facebook.presto.spi.ErrorCodeSupplier;
+
+import static com.facebook.presto.common.ErrorType.EXTERNAL;
+import static com.facebook.presto.common.ErrorType.INTERNAL_ERROR;
+
+public enum ArrowErrorCode
+        implements ErrorCodeSupplier
+{
+    ARROW_INVALID_TABLE(0, EXTERNAL),
+    ARROW_INVALID_CREDENTAILS(1, EXTERNAL),
+    ARROW_FLIGHT_ERROR(2, EXTERNAL),
+    ARROW_INTERNAL_ERROR(3, INTERNAL_ERROR);
```

Comment:
All but `ARROW_FLIGHT_ERROR` are unused.

I think `ARROW_FLIGHT_ERROR` is too generic.  Let's think of sensible error codes.

Reply:
created additional error codes for different scenarios.